src/components/Navbar/SearchRepo/SearchRepo.tsx

```diff
@@ -1,35 +1,30 @@
 import * as React from "react";
 import get from "lodash/get";
+import debounce from "lodash/debounce";
 import { AutoComplete } from "antd";
+import { BaseSelectRef } from "rc-select";
 import { useLazyQuery } from "@apollo/client";
-import { SearchReposDocument, SearchReposQuery } from "./search-repo.gql.generated";
+import { SearchReposDocument } from "./search-repo.gql.generated";
 import { SearchType } from "@/generated/github";
 import { repoState } from "@/local-state";
 
 function SearchRepo() {
-  const delay = React.useRef<NodeJS.Timeout>();
-  const searchRef = React.useRef<any>();
-  const [searchRepos, { data }] = useLazyQuery<SearchReposQuery>(SearchReposDocument);
+  const searchRef = React.useRef<BaseSelectRef | null>(null);
+  const [searchRepos, { data }] = useLazyQuery(SearchReposDocument);
 
-  const handleSearch = (value: string) => {
-    if (delay.current) {
-      clearTimeout(delay.current);
-    }
-
-    delay.current = setTimeout(() => {
-      searchRepos({
-        variables: {
-          query: value,
-          type: SearchType.REPOSITORY,
-          first: 20,
-        },
-      });
-    }, 500);
-  };
+  const handleSearch = debounce((value: string) => {
+    searchRepos({
+      variables: {
+        query: value,
+        type: SearchType.REPOSITORY,
+        first: 20,
+      },
+    });
+  }, 500);
 
   const handleSelect = (value: string) => {
     const repo = data?.search?.edges?.find(
-      (item) => get(item, "node.nameWithOwner", "") === value
+      (item) => get(item, "node.nameWithOwner") === value
```

Comment:
Chỗ này mình định khuyên không nên dùng `get` do nó sẽ cho ra type là `any`. Nếu được thì mình vẫn nên xử lý đúng type, còn không thì dùng typecast:
```suggestion
      (item) => get(item, "node.nameWithOwner", "") as string === value
```